httpbp/client_middlewares.go

```diff
@@ -467,7 +468,15 @@ func (c clientFaultMiddleware) Middleware() ClientMiddleware {
 			address := req.URL.Hostname()
 			method := strings.TrimPrefix(req.URL.Path, "/")
 			header := httpHeader(req.Header)
-			return c.injector.InjectWithAbortOverride(req.Context(), address, method, &header, resume, abort)
+			return c.injector.Inject(req.Context(),
+				faults.InjectParameters[*http.Response]{
+					Address:     address,
+					Method:      method,
+					MethodLabel: "",
```

Comment:
🔕 this is under `internal` api so we can change this later, thus 🔕, but this is going to be a huge source of confusion in the future.

Reply:
Ack. I'll think of a better long-term solution for a follow-up PR then.